papermill/iorw.py

```diff
@@ -88,6 +88,11 @@ def listdir(cls, path):
 
     @classmethod
     def write(cls, buf, path):
+        dir = os.path.dirname(path)
+        if not dir.endswith("/"):
+            dir += "/"
+        if not os.path.exists(dir):
+            raise NoSuchDirectory('output folder {} doesn\'t exist.'.format(dir))
```

Comment:
Let's still use FileNotFoundError instead of making a new error class, as it's used for missing directories as well.

From Python docs:
| exception FileNotFoundError
| Raised when a file or directory is requested but doesn’t exist. Corresponds to errno ENOENT.

Reply:
@MSeal used FileNotFoundError for throwing the error if a folder or file doesn't exist but the build is failing because FileNotFoundError is not directly supported in Python 2.7. Could you please let me know if we can proceed this way.